lib/api/_loaders/_base-loader.js

```diff
@@ -237,47 +238,120 @@ class BaseLoader extends EventEmitter {
 
   validateMethod() {}
 
-  defineArgs(parent) {
+  defineArgs(parent, namespacedApi) {
     const {commandName} = this;
     const commandFn = this.commandFn.bind(this);
+
+    let stringifiedNamespace;
+    let mainNamespace;
+    let apiToReturn;
+
+    if (this.namespace && Utils.isString(this.namespace)) {
+      stringifiedNamespace = this.namespace;
+      mainNamespace = this.namespace;
+    } else if (Array.isArray(this.namespace) && this.namespace.length > 0) {
+      stringifiedNamespace = this.namespace.join('.');
+      mainNamespace = this.namespace[0];
+    }
+
+    if (namespacedApi && mainNamespace) {
+      namespacedApi[mainNamespace] = namespacedApi[mainNamespace] || {};
+      apiToReturn = new Proxy(namespacedApi[mainNamespace], {
+        get(_, name) {
+          // namespacedApi[mainNamespace] may change after API loading is finished.
+          return namespacedApi[mainNamespace][name];
+        }
+      });
+    }
+
     const args = [this.createQueuedCommandFn({
       parent,
       commandName,
+      namespace: stringifiedNamespace,
       commandFn,
+      apiToReturn,
       context: this
     })];
 
-    const namespace = this.getTargetNamespace(parent);
+    const namespace = this.getTargetNamespace(parent, namespacedApi);
     if (namespace) {
       args.unshift(namespace);
     }
 
     return args;
   }
 
+  isRestrictedNamespace(parent, namespace) {
+    if (!parent) {
+      return false;
+    }
+
+    // do not load these namespaces on parent (page objects and within-context).
+    const namespacesRestrictedForParent = ['client'];
+
+    const isRestrictedNamespace = namespace
+      && namespacesRestrictedForParent.some((ns) => namespace.toString().startsWith(ns));
+
+    return isRestrictedNamespace;
+  }
```

Comment:
This change here is for the discussion we had that we don't need to load all the namespaces directly onto the page objects but only a few like `cookies` and `a11y`. So, with this, we can restrict the namespaces we do not want to load on page-objects.

This was actually added here because page objects already had a `client` property (set to `NightwatchClient`) and so we could not add the `client` namespace commands directly onto page-objects, but it also serves the purpose mentioned above. And since we are no longer loading the commands onto `client` namespace, we can safely remove this as well if we do not want this feature at this point.